src/MF_Modules/MFOutputShifter.cpp

```diff
@@ -46,49 +53,52 @@ void MFOutputShifter::attach(uint8_t latchPin, uint8_t clockPin, uint8_t dataPin
   _moduleCount = moduleCount;
 
   pinMode(_latchPin, OUTPUT);
-  pinMode(_clockPin, OUTPUT);  
+  pinMode(_clockPin, OUTPUT);
   pinMode(_dataPin, OUTPUT);
 
   clear();
 }
 
 void MFOutputShifter::detach()
 {
-  if (!_initialized) return;
+  if (!_initialized)
+    return;
   _initialized = false;
 }
 
-void MFOutputShifter::clear() 
+void MFOutputShifter::clear()
 {
   // Set everything to 0
-  for (uint8_t i = 0; i < _moduleCount; i++) {   
+  for (uint8_t i = 0; i < _moduleCount; i++)
+  {
     _outputBuffer[i] = 0;
   }
-  updateShiftRegister();  
+  updateShiftRegister();
 }
 
-
-void MFOutputShifter::test() 
+void MFOutputShifter::test()
 {
-  for (uint8_t b = 0; b < _moduleCount * 8; b++) {   
-    uint8_t idx = (b & 0xF8)>>3;
+  for (uint8_t b = 0; b < _moduleCount * 8; b++)
+  {
+    uint8_t idx = (b & 0xF8) >> 3;
     uint8_t msk = (0x01 << (b & 0x07));
     _outputBuffer[idx] = msk;
-    if(msk == 0x01 && idx != 0) _outputBuffer[idx-1] = 0x00;
+    if (msk == 0x01 && idx != 0)
+      _outputBuffer[idx - 1] = 0x00;
     updateShiftRegister();
     delay(50);
 
-    _outputBuffer[_moduleCount-1] = 0;
+    _outputBuffer[_moduleCount - 1] = 0;
     updateShiftRegister();
   }
 }
 
-
 void MFOutputShifter::updateShiftRegister()
 {
-   digitalWrite(_latchPin, LOW);
-   for (uint8_t i = _moduleCount; i>0 ; i--) {
-      shiftOut(_dataPin, _clockPin, MSBFIRST, _outputBuffer[i]); //LSBFIRST, MSBFIRST,
-   }    
-   digitalWrite(_latchPin, HIGH);
+  digitalWrite(_latchPin, LOW);
+  for (uint8_t i = _moduleCount; i > 0; i--)
+  {
+    shiftOut(_dataPin, _clockPin, MSBFIRST, _outputBuffer[i - 1]); // LSBFIRST, MSBFIRST,
```

Comment:
This is the actual change. All the rest is whitespace formatting done by VSCode to clean it up.

Reply:
either we share all the same formatting rules or you should not reformat everything if there is no need.

your formatting is not necessarily my preference, e.g. i like the curly braces in the same line where the if is. But most importantly it should be aligned with everyone.

Reply:
I'm not doing anything intentionally, it's just VSCode doing it. I wonder if I have a formatter installed? I'll check.

Reply:
I think it's probably part of the VSCode C++ extensions? I don't see any explicit formatter, but it's definitely changing things on save. I'm trying to update the PR with just the code change, hang tight.

Reply:
In order to avoid frequent style discrepancies, we could conveniently use the automatic formatter of the "C/Cpp" plugin of VSCode (in `settings.json`).
If you want to give it a try, my settings (slightly modified from the defaults) are:
`"C_Cpp.clang_format_fallbackStyle": "{ BasedOnStyle: LLVM, UseTab: Never, IndentWidth: 4, TabWidth: 4, BreakBeforeBraces: Linux, AllowShortIfStatementsOnASingleLine: false, IndentCaseLabels: false, ColumnLimit: 0, AccessModifierOffset: -4, NamespaceIndentation: All, FixNamespaceComments: false, AlignConsecutiveAssignments: AcrossComments, AlignConsecutiveBitFields: AcrossEmptyLinesAndComments, AlignConsecutiveDeclarations: AcrossEmptyLinesAndComments, AlignConsecutiveMacros: AcrossComments, AlignTrailingComments: true, AllowShortBlocksOnASingleLine: Empty, AllowShortEnumsOnASingleLine: true, AllowShortFunctionsOnASingleLine: Inline, AllowShortIfStatementsOnASingleLine: WithoutElse,  BreakInheritanceList: BeforeColon, EmptyLineBeforeAccessModifier: LogicalBlock, IndentPPDirectives: BeforeHash }"
`

Reply:
I fully support an automatic formatter that everyone has access to via VSCode. I don't care at all what the settings are, I just want to know when I hit "save" things will format to however is the right way to be consistent with everyone else.